### DISCUSSION TITLE
RecursionError: maximum recursion depth exceeded while pickling an object

### DISCUSSION THREAD

**Discussion by AILUOMANGALAOSHI** [networkx/networkx#6781] [Q&A]
I use function write_gpickle to save a graph with 6032 nodes.get a error 
pickle.dump(G, path, protocol)
RecursionError: maximum recursion depth exceeded while pickling an object

**Comment by dschult** [comment]
I don't get this error. Can you provide a small example where it happens? What other code are you using that might make the graph use code with recursion involved? For example are you saving attributes on the graph which depend on attributes on the graph, etc.

**Comment by rossbar** [comment]
`nx.write_gpickle` was deprecated back in NX version 2.6 and has been removed since v3.0. The recommendation is [to use pickle directly instead](https://networkx.org/documentation/stable/release/migration_guide_from_2.x_to_3.0.html#deprecated-code)
